lib/matplotlib/axes/_base.py

```diff
@@ -3394,6 +3394,8 @@ def tick_params(self, axis='both', **kwargs):
             Width of gridlines in points.
         grid_linestyle : str
             Any valid `.Line2D` line style spec.
+        gridOn : bool
```

Comment:
I'm hesitant to further expand the use of camelcase variables. But I don't have an overview of its usage and whether `grid_on` would be a viable alternative. (No code editor access right now).

Reply:
`grid_on` would be available as an alternative. `gridOn` is only used internally in 4 files, no big deal to rename.  

As an alternative one could translate `gridOn` to `grid_on` (which wouldn't, however, prevent anyone to use the then undocumented keyword in `set_tick_params` etc., as it's the case now).  

If we don't want to have `gridOn` I'd opt for renaming it to `grid_on`. This wouldn't be an API change as `gridOn` is not documented.